cmake/helpers.cmake

```diff
@@ -76,3 +76,9 @@ function(FetchContentSparse_Declare name GIT_REPOSITORY GIT_TAG GIT_DIR)
         WORKING_DIRECTORY ${content-build-dir})
     FetchContent_Declare(${name} SOURCE_DIR ${content-build-dir}/${GIT_DIR})
 endfunction()
+
+# Escape a string to be able to be processed by REGEX
+function(EscapeRegex IN_OUT_STR)
+    string(REGEX REPLACE "([][+.*()^])" "\\\\\\1" ${IN_OUT_STR} "${${IN_OUT_STR}}")
```

Comment:
Python has a `re.escape` function. I think it'd be easier to use python instead of torturing ourselves by trying to cobble something together in CMake.

Reply:
Yeah I can look into that. It's a bit strange the way this works, the `\\\\\\1` becomes `\\\1` after the cmake language processing. The first `\\` places the escaping `\` in the output and the `\1` is a placeholder for the match.